File: z-clients/javascript/src/models/msgQueueExtendLeaseIn.ts
// this file is @generated
import { Temporal } from 'temporal-polyfill-lite';

export interface MsgQueueExtendLeaseIn {
    namespace?: string | null;
    msgIds: string[];
    leaseDuration?: Temporal.Duration;
}

export interface MsgQueueExtendLeaseIn_ {
    namespace?: string | null;
    topic: string;
    consumerGroup: string;
    msgIds: string[];
    leaseDuration?: Temporal.Duration;
}

export const MsgQueueExtendLeaseInSerializer = {
    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _fromJsonObject(object: any): MsgQueueExtendLeaseIn_ {
        return {
            namespace: object['namespace'],
            topic: object['topic'],
            consumerGroup: object['consumer_group'],
            msgIds: object['msg_ids'],
            leaseDuration: object['lease_duration_ms'] != null ? Temporal.Duration.from({ milliseconds: object['lease_duration_ms'] }) : undefined,
        };
    },

    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _toJsonObject(self: MsgQueueExtendLeaseIn_): any {
        return {
            'namespace': self.namespace,
            'topic': self.topic,
            'consumer_group': self.consumerGroup,
            'msg_ids': self.msgIds,
            'lease_duration_ms': self.leaseDuration != null ? self.leaseDuration.total('millisecond') : undefined,
        };
    }
}

File: z-clients/python/diom/models/msg_queue_extend_lease_in.py
# this file is @generated
import typing as t
from pydantic import Field

from ..internal.base_model import BaseModel
from ..internal.types import TimeDeltaMs


class MsgQueueExtendLeaseIn(BaseModel):
    namespace: str | None = None

    msg_ids: t.List[str]

    lease_duration: TimeDeltaMs | None = Field(alias="lease_duration_ms", default=None)


class _MsgQueueExtendLeaseIn(BaseModel):
    namespace: str | None = None

    topic: str

    consumer_group: str

    msg_ids: t.List[str]

    lease_duration: TimeDeltaMs | None = Field(alias="lease_duration_ms", default=None)
